.gitignore

```diff
@@ -1,3 +1,8 @@
+# ignore generated templates
+templates/*.yaml
+# templates/namespace.yaml is not generated, it's a special case
+!templates/namespace.yaml
+
```

Comment:
This looks similar to the `.gitignore` lines inside the the [charts](https://github.com/open-feature/open-feature-operator/blob/main/chart/open-feature-operator/.gitignore)

Are they needed? Why would there be a `/template` dir here?

Reply:
Looks like I confused myself on the latest rounds of conflicts, I have resolved this